Casks/g/garmin-express.rb

```diff
@@ -1,17 +1,30 @@
 cask "garmin-express" do
-  version "7.22.1.0,7220100"
-  sha256 :no_check
+  on_catalina :or_older do
+    version "7.18.0"
+    sha256 "f032349fa2b4cd5abe20ae9201c905874a4d3ef1b328966a29d24e6e15793729"
+
+    url "https://download.garmin.com/omt/express/GarminExpress_#{version}_OS10x.dmg"
+
+    livecheck do
+      skip "Legacy version"
+    end
+  end
+  on_big_sur :or_newer do
+    version "7.22.1"
+    sha256 :no_check
+
+    url "https://download.garmin.com/omt/express/GarminExpress.dmg"
+
+    livecheck do
+      url "https://support.garmin.com/capi/faq/details/?locale=en-US&faqId=9MuiEv9c2y2wgcXvzEVEe8"
+      regex(/for\s+Mac:.*?v?(\d+(?:\.\d+)+)/i)
```

Comment:
```suggestion
      regex(/for\s+Mac:?[\s\u00a0]*?v?(\d+(?:\.\d+)+)/i)
```

The intention of `.*?` appears to be to account for the whitespace after `:` in text like "Current Version of Garmin Express for Mac:  7.22.1 as of June 6, 2024" but the looseness could lead to unintended matches in the future. Ideally we would use `\s*` here but I see that there's a no-break space (`\u00A0`) after the colon that prevents it from matching. That said, `[\s\u00a0]*?` seems to work.